src/main/java/eu/xenit/alfred/content/gateway/collections/ObservableMap.java

```diff
@@ -0,0 +1,122 @@
+package eu.xenit.alfred.content.gateway.collections;
+
+import static reactor.core.publisher.Sinks.EmitFailureHandler.FAIL_FAST;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import lombok.Value;
+import lombok.experimental.Delegate;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+
+
+public class ObservableMap<K, V> implements Map<K, V>, AutoCloseable {
+
+    @Delegate(types = MapReadOperations.class)
+    private Map<K, V> map = new LinkedHashMap<>();
+    private final Sinks.Many<MapUpdate<K, V>> updates = Sinks.many().multicast().onBackpressureBuffer();
+
+
+    @Override
+    public V get(Object key) {
+        return this.map.get(key);
+    }
+
+    public V put(K key, V value) {
+        V previousValue = map.put(key, value);
+        updates.emitNext(MapUpdate.put(key, value), FAIL_FAST);
+        return previousValue;
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> map) {
+        map.forEach(this::put);
+    }
+
+    @Override
+    public void clear() {
+        this.map.clear();
+        updates.emitNext(MapUpdate.clear(), FAIL_FAST);
+    }
+
+    @Override
+    public V remove(Object key) {
+        V removedValue = map.remove(key);
+        updates.emitNext(MapUpdate.remove(key, removedValue), FAIL_FAST);
+        return removedValue;
+    }
+
+    public Flux<MapUpdate<K, ? extends V>> observe() {
+        return Flux.concat(
+                Flux.fromIterable(map.entrySet())
+                        .map(entry -> new MapUpdate<>(UpdateType.PUT, entry.getKey(), entry.getValue())),
+                updates.asFlux()
+        );
+    }
+
+    @Override
+    public void close() {
+        this.updates.emitComplete(FAIL_FAST);
+    }
+
+    public enum UpdateType {
+        PUT, REMOVE, CLEAR
+    }
+
+    @Value
+    public static class MapUpdate<K, V> {
+
+        UpdateType type;
+        Object key;
```

Comment:
Why is this `Object` instead of  `K`?

Reply:
As discussed in other comments: because of the contract `Map.get(Object key)`